packages/fxa-settings/src/models/integrations/oauth-native-integration.ts

```diff
@@ -113,15 +127,34 @@ export class OAuthNativeIntegration extends OAuthWebIntegration {
   }
 
   wantsKeys() {
+    // TODO: this will not always be true when working on FXA-12374
     return true;
   }
 
-  // TODO in FXA-10313, check for "Relay" or whatever makes sense at implementation
-  get serviceName() {
-    if (this.data.service === 'sync') {
+  getWebChannelServices(syncEngines?: SyncEngines) {
+    if (this.isFirefoxClientServiceRelay()) {
+      return { relay: {} };
+    }
+    if (this.isFirefoxClientServiceAiMode()) {
+      return { aimode: {} };
+    }
+    if (this.isDefaultSyncService()) {
+      return { sync: syncEngines || {} };
+    }
+    return undefined;
+  }
+
+  getServiceName() {
```

Comment:
I noticed we weren't using `serviceName` for anything. This does overwrite the `getServiceName` method we've got [in the base integration](https://github.com/mozilla/fxa/blob/83e0d4be5feeef6ef7d07e33ce762f7a4f70469c/packages/fxa-settings/src/models/integrations/integration.ts#L105), but that looked OK to me.

Now if you run `yarn firefox` to pull up this flow and you change the `service` param to `service=aimode` you'll see this, though it's currently the only UI difference in this flow which is noted in the ticket:
<img width="653" height="549" alt="image" src="https://github.com/user-attachments/assets/0b8727d3-f791-4c5d-b0fb-13fbd3d6c8f4" />